### ISSUE TITLE
Part of metadata file is deprecated 

### ISSUE THREAD

**Issue by BKaraargirov** [scikit-image/scikit-image#6001]
## Description
The METADATA generated by setup.py has deprecated REQUIRES fields example from 0.18.3:
```
Requires: numpy (>= 1.16.5)
Requires: scipy (>= 1.0.1)
Requires: matplotlib (>= 2.0.0,!=3.0.0)
Requires: networkx (>= 2.0)
Requires: pillow (>= 4.3.0,!=7.1.0,!=7.1.1)
Requires: imageio (>= 2.3.0)
Requires: tifffile (>= 2019.7.26)
Requires: PyWavelets (>= 1.1.1)
Requires-Python: >=3.7
Description-Content-Type: text/markdown
Requires-Dist: numpy (>=1.16.5)
Requires-Dist: scipy (>=1.0.1)
Requires-Dist: matplotlib (!=3.0.0,>=2.0.0)
Requires-Dist: networkx (>=2.0)
Requires-Dist: pillow (!=7.1.0,!=7.1.1,>=4.3.0)
Requires-Dist: imageio (>=2.3.0)
Requires-Dist: tifffile (>=2019.7.26)
```

This causes a warning in tools such as PEX: 
```
PEXWarning: Ignoring 8 `Requires` fields in scikit-image 0.18.3 metadata:
1.) Requires: numpy (>= 1.16.5)
2.) Requires: scipy (>= 1.0.1)
3.) Requires: matplotlib (>= 2.0.0,!=3.0.0)
4.) Requires: networkx (>= 2.0)
5.) Requires: pillow (>= 4.3.0,!=7.1.0,!=7.1.1)
6.) Requires: imageio (>= 2.3.0)
7.) Requires: tifffile (>= 2019.7.26)
8.) Requires: PyWavelets (>= 1.1.1)

You may have issues using the 'scikit-image' distribution as a result.
More information on this workaround can be found here:
  https://github.com/pantsbuild/pex/issues/1201#issuecomment-791715585
```
Seeing as there is both requires and requires-dist(which is supposed to be used from metadata v1.2 onward ) is there still need to have both ? 

I think this happens because both requires and install_requires parameters are used: https://github.com/scikit-image/scikit-image/blob/b566789687b2ff575fa21189d3a8d7b5fa0af287/setup.py#L243

## Version information

python 3.7.9 (default, Jun 10 2021, 14:08:03)
[Clang 12.0.5 (clang-1205.0.22.9)]
system: Darwin-20.6.0-x86_64-i386-64bit
scikit-image version: 0.18.3
numpy version: 1.18.0

**Comment by grlee77** [comment]
Thanks for reporting this. If you want to play around and see what fixes it locally, that would be helpful. Otherwise, we will try to take a look soon

**Comment by grlee77** [comment]
closed by #6017
